### ISSUE TITLE
fuse non-zero exit code

### ISSUE THREAD

**Issue by magikid** [solo-io/packer-builder-arm-image#57]
When I try to build my image using this builder with the command `sudo packer build image.json`, the build fails with the error message

```
==> arm-image: fuser -k /tmp/748879671
Build 'arm-image' errored: exit status 1

==> Some builds didn't complete successfully and had errors:
--> arm-image: exit status 1

==> Builds finished but no artifacts were created.
```

I looked into the exit status of `fuser` and if no one is accessing the file, the exit code is 1 so I think there might be a logic flaw in here somewhere.

**Comment by magikid** [comment]
I've narrowed the issue down to this provisioner in my JSON:

```
{                                                                                                                       
  "type": "file",                                                                                                       
  "source": "public-ssh-key.pub",                                                                                       
  "destination": "/home/pi/.ssh/authorized_keys"                                                                        
}
```

If I remove that, the image builds successfully.

**Comment by magikid** [comment]
Ah, I think I discovered the issue.  According to the provisioner docs (emphasis mine):
> This value must be a writable location and any parent directories **must already exist**.

I didn't bother to create the containing folders before copying them so I think that's why it was failing.  I also went ahead and specified the full path for the source file.  I'm building with docker so everything's mounted at `/build`.  The final block that works for me is:
```
    {
      "type": "file",
      "source": "/build/public-ssh-key.pub",
      "destination": "/home/pi/.ssh/authorized_keys"
    },
```
